### PR TITLE
Feature/environment

### PR DESCRIPTION
**Pull request by technetos**
Adds basic environment status to a project

### CODE REVIEW

File: C++/Env/Dev.h [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#ifndef _DEV_H_
```
Comment (DannyPeck) [inline C++/Env/Dev.h]: do we want to follow the convention of `_MACRO_` or `MACRO`?

File: C++/Env/Dev.h [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+class Dev : public Env
+{
+public:
+  std::string getEnv (void);
+
+private:
+  constexpr static const char * env = "DEVELOPMENT";
+};
+
+#endif
```
Comment (DannyPeck) [inline C++/Env/Dev.h]: I find #endif // MACRO to be a good convention as it shows the end of the associated ifndef statement

File: C++/Env/Env.h [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+class Env
+{
+public:
+  virtual std::string getEnv (void) = 0;
+  
+private:
+};
+
+#endif
```
Comment (DannyPeck) [inline C++/Env/Env.h]: see comment above

File: C++/Env/Env.h [hunk 4]
```diff
@@ -0,0 +1,14 @@
+#ifndef _ENV_H_
```
Comment (DannyPeck) [inline C++/Env/Env.h]: see comment above

File: C++/Env/Prod.cpp [hunk 5]
```diff
@@ -0,0 +1,6 @@
+#include "Prod.h"
+
+std::string Prod::getEnv (void)
+{
+  return std::string { env };
```
Comment (DannyPeck) [inline C++/Env/Prod.cpp]: Do we like the spacing between the { }?

File: C++/Env/Prod.h [hunk 6]
```diff
@@ -0,0 +1,15 @@
+#ifndef _PROD_H_
```
Comment (DannyPeck) [inline C++/Env/Prod.h]: see comment above

File: C++/Env/Prod.h [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+class Prod : public Env
+{
+public:
+  std::string getEnv (void);
+
+private:
+  constexpr static const char * env = "PRODUCTION";
+};
+
+#endif
```
Comment (DannyPeck) [inline C++/Env/Prod.h]: see comment above

**Review by DannyPeck — COMMENTED** [review]
Looks like a solid start, just a few conversational comments.  I personally think we should elongate the naming to be Production, Development, Environment, etc.

File: C++/Env/Dev.h (hunk 1 above)
Comment (technetos) [reply C++/Env/Dev.h]: `_MACRO_` for ifndef's
`MACRO` for actual macros that are used for other purposes

File: C++/Env/Prod.cpp (hunk 5 above)
Comment (technetos) [reply C++/Env/Prod.cpp]: yes

File: C++/Env/Environment.h [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+#define _ENVIRONMENT_H_
+
+#include <string>
+
+class Environment
+{
+public:
+  virtual std::string getEnvironment (void) = 0;
+  
+private:
```
Comment (DannyPeck) [inline C++/Env/Environment.h]: We can get rid of this at the moment.

**Review by DannyPeck — APPROVED** [review]
Other than that one small thing, it looks great!  Feel free to merge it after addressing that issue.

File: C++/Env/Environment.h (hunk 8 above)
Comment (technetos) [inline C++/Env/Environment.h]: explain please

Comment (technetos) [reply C++/Env/Environment.h]: No we cant.  How else would you like to call this method on the derived classes?

**Pull request merged**